x/mongo/driver/auth/creds/gcpcreds.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) MongoDB, Inc. 2022-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package creds
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"go.mongodb.org/mongo-driver/internal"
+)
+
+// GcpCredentialProvider provides GCP credentials.
+type GcpCredentialProvider struct {
+	HTTPClient *http.Client
+}
+
+// GetCredentials generates GCP credentials.
+func (p *GcpCredentialProvider) GetCredentials(ctx context.Context) (string, error) {
+	metadataHost := "metadata.google.internal"
```

Comment:
```suggestion
	const metadataHost = "metadata.google.internal"
```

Consider defining this as a constant.

Reply:
`metadataHost` is not a constant. It may be updated at L29 by environment variable "GCE_METADATA_HOST".